applications/StructuralMechanicsApplication/custom_response_functions/response_utilities/linear_strain_energy_response_function.h

```diff
@@ -0,0 +1,195 @@
+// KRATOS  ___|  |                   |                   |
+//       \___ \  __|  __| |   |  __| __| |   |  __| _` | |
+//             | |   |    |   | (    |   |   | |   (   | |
+//       _____/ \__|_|   \__,_|\___|\__|\__,_|_|  \__,_|_| MECHANICS
+//
+//  License:		 BSD License
+//					 license: structural_mechanics_application/license.txt
+//
+//  Main authors:    Armin Geiser, https://github.com/armingeiser
+//
+
+#if !defined(KRATOS_LINEAR_STRAIN_ENERGY_H_INCLUDED )
+#define  KRATOS_LINEAR_STRAIN_ENERGY_H_INCLUDED
+
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "includes/model_part.h"
+#include "includes/kratos_parameters.h"
+#include "structural_mechanics_application_variables.h"
+
+namespace Kratos {
+
+///@name Kratos Globals
+///@{
+
+///@}
+///@name Type Definitions
+///@{
+
+///@}
+///@name  Enum's
+///@{
+
+///@}
+///@name  Functions
+///@{
+
+///@}
+///@name Kratos Classes
+///@{
+
+/// Short class definition.
+class KRATOS_API(STRUCTURAL_MECHANICS_APPLICATION) LinearStrainEnergyResponseFunction {
+public:
+
+    ///@name Type Definitions
+    ///@{
+
+    ///@}
+    ///@name Pointer Definitions
+    /// Pointer definition of LinearStrainEnergyResponseFunction
+    KRATOS_CLASS_POINTER_DEFINITION(LinearStrainEnergyResponseFunction);
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    LinearStrainEnergyResponseFunction(ModelPart& rModelPart, Parameters ResponseSettings);
+
+    /// Destructor.
+    virtual ~LinearStrainEnergyResponseFunction() {
+    }
+
+    ///@}
+    ///@name Operators
+    ///@{
+
+    ///@}
+    ///@name Operations
+    ///@{
+
+    double CalculateValue();
```

Comment:
The other functions are virtual so if the intention is to derive from this class, this function should probably also be virtual.